app/src/main/java/ch/epfl/polybazaar/FillListingActivity.java

```diff
@@ -225,7 +248,9 @@ private void submit() {
                 }
             };
             String category = spinnerList.get(spinnerList.size()-1).getSelectedItem().toString();
-            Listing newListing = new Listing(titleSelector.getText().toString(), descriptionSelector.getText().toString(), priceSelector.getText().toString(), "test.user@epfl.ch", stringImage, category);
+            FirebaseAuthenticator fbAuth = FirebaseAuthenticator.getInstance();
+            String userEmail = fbAuth.getCurrentUser() == null ? "NO_USER@epfl.ch" : fbAuth.getCurrentUser().getEmail();
```

Comment:
Is there any advantage of having "NO_USER@epfl.ch" instead of null here ? It seems that the situation where the user is null is rather unexpected and should probably fail right away.

Reply:
You are right this behaviour is weird. I added it because a bunch of previous tests use the submit method without logging in before and fail without this line.
I'll add a TODO and the person refactoring the code next week may refactor these tests